tiledb/sm/query/query.cc

```diff
@@ -1359,8 +1035,9 @@ Status Query::set_data_buffer(
       return set_coords_buffer(buffer, buffer_size);
   }
 
-  const bool is_var = array_schema_->var_size(name);
-  if (is_dim && !is_var &&
+  const bool is_var = is_aggr ? default_channel_aggregates_[name]->var_sized() :
+                                array_schema_->var_size(name);
+  if (!is_aggr && is_dim && !is_var &&
```

Comment:
This logic confuses me: "not x and y and not z and other stuff". I'm ignoring the `other stuff` aspect because that's obvious. And back to my naming collision thing, we have three booleans which gives an 8 element boolean matrix and I have no idea what any of those combinations really means.